### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
 - Use Webpack for easy AMD to ES6 transition. It's the best of both worlds! Webpack is able to consume AMD, CommonJS, or ES6 module sources and export them as a nice UMD package for legacy clients (which is everything for the moment, but this will make it easy for us to start to use ES6 features where it makes sense).
 - Import individual Lodash functions (as CommonJS modules) from NPM.
 - Quick pass at ES6-ify Neue scripts.
 - Remove outdated code style guidelines.

For review: @DoSomething/front-end

### CODE REVIEW

**Comment by weerd** [comment]
Exciting! Looks good to me. Will need to text a bit, but liking the Webpack stuffs!

**Comment by DFurnes** [comment]
Made a few bug fixes, and tested that this continues to work in AMD build system in the main app.

File: Gruntfile.js [hunk 1]
```diff
@@ -155,7 +111,79 @@ module.exports = function(grunt) {
...
+          library: 'Neue',
+          libraryTarget: 'umd'
+        },
+        externals: {
+          // Don't bundle the 'jquery' package in neue.js, but
+          // instead load from `jQuery` global variable or AMD/CJS package.
+          'jquery': {
+            root: 'jQuery',
+            commonjs: 'jquery',
+            commonjs2: 'jquery',
```
Comment (weerd) [inline Gruntfile.js]: what's this? just curious

Comment (DFurnes) [inline Gruntfile.js]: See [docs](http://webpack.github.io/docs/configuration.html#output-librarytarget)! Actually just realized I can safely remove plain-ol' `commonjs` from this object too – `umd` exports to AMD, CommonJS2 or as property in root.

**Comment by weerd** [comment]
:+1:

**Comment by DFurnes** [comment]
Weeeeeeeird. This merged (see @3b21b3f9e8c4cdca1c46207d31c09f2f667bab1f), but Github doesn't seem to have realized. Closing.

**Pull request closed without merging**